pkg/validator/cluster.go

```diff
@@ -312,13 +312,79 @@ func validateGeneralClusterChanges(
 	valErrors []string,
 ) []string {
 
-	if cr.Spec.AppID != prevCr.Spec.AppID {
-		appModifiedMsg := fmt.Sprintf(
-			modifiedProperty,
-			"app",
+	readyToUpgrade := shared.ClusterIsReady(prevCr)
```

Comment:
Since this will cover both upgrade and rollback readiness, let's call the variable something else like `okToChangeApp` or something.

Reply:
I meant this variable covers any cluster changes readiness, including pods count changing. Firstly it checks, if cluster is `configured`, and then, also checks if a user requested the rollback.

For the rollback case there is an additional check below, inside the `if appModified` block, that the previous and current cluster specs have no differences other than `AppID`